### PR TITLE
add jacobian

### PR DESCRIPTION
**Pull request by sangyx**
I have completed the function. I will add test cases later.

### CODE REVIEW

File: src/udiff/_vjp_diffs.py [hunk 1]
```diff
@@ -553,12 +553,12 @@ def matmul_adjoint_1(A, G, A_ndim, B_meta):
 
 def matmul_vjp_0(ans, A, B):
     A_meta = metadata(A)
-    B_ndim = np.ndim(B)
+    B_ndim = B.ndim
```
Comment (hameerabbasi) [inline src/udiff/_vjp_diffs.py]: Why are these changes necessary?

Comment (sangyx) [reply src/udiff/_vjp_diffs.py]: If we use `np.ndim`, it will return a `DiffArray`, which will cause an error. There may be a bug. I'm trying to figure out how to fix it.

Comment (sangyx) [reply src/udiff/_vjp_diffs.py]: The error is caused by `return NotImplemented`. When we call `np.ndim(B)`, `B` is a raw `np.array`, and `coerce` is `False`, so the function `__ua_convert__` return `NotImplemented`.

Comment (hameerabbasi) [reply src/udiff/_vjp_diffs.py]: Can you provide a script?

Comment (sangyx) [reply src/udiff/_vjp_diffs.py]: I have fixed it. The error is caused by the code in `register_vjp`:
```python
self._vjp = vjpmaker(tuple(parent_argnums), self.value, tuple(vjp_args), kwargs)
```

**Comment by sangyx** [comment]
Could you merge this pr or give more comments?

**Comment by hameerabbasi** [comment]
Oh, I apologise. I didn’t know you were done working on it.

File: src/udiff/_vjp_diffs.py (hunk 1 above)
Comment (hameerabbasi) [reply src/udiff/_vjp_diffs.py]: This is still using `.ndim` instead of `np.ndim`. Did you forget to push?

File: src/udiff/_vjp_diffs.py [hunk 2]
```diff
@@ -553,12 +553,12 @@ def matmul_adjoint_1(A, G, A_ndim, B_meta):
...
     return lambda g: matmul_adjoint_0(B, g, A_meta, B_ndim)
 
 
 def matmul_vjp_1(ans, A, B):
-    A_ndim = np.ndim(A)
+    A_ndim = A.ndim
```
Comment (hameerabbasi) [inline src/udiff/_vjp_diffs.py]: Same here.

**Review by hameerabbasi — COMMENTED** [review]
I think you forgot to push.

File: src/udiff/_vjp_diffs.py (hunk 2 above)
Comment (sangyx) [reply src/udiff/_vjp_diffs.py]: I think there is no difference between these two lines. Is it better to use `np.ndim`?

Comment (hameerabbasi) [reply src/udiff/_vjp_diffs.py]: If you’re sure this is a `DiffArray`, then it’s the same, otherwise the other package might not have implemented `.ndim` in which case the `np.ndim` will always do the right thing.
